### DISCUSSION TITLE
SBUS pin on FSI6X?

### DISCUSSION THREAD

**Discussion by FRPProject** [OpenI6X/opentx#193] [Q&A]
**_Do you know which is the pin that outputs SBUS on FSI6X? is it PA9? I am trying to use SBUS pin on the FSI6X board for my external module. Thank you._** 

![image](https://user-images.githubusercontent.com/37748115/157159386-d2a6d94e-72e8-420c-b788-f8b3908e9499.png)

![image](https://user-images.githubusercontent.com/37748115/157159409-b28001b2-b082-496b-b1d3-1b60d8b03f6c.png)

**Comment by ajjjjjjjj** [comment]
It's not implemented currently as no one of us have SBUS module. This functionality is waiting for volunteers.

**Reply by FRPProject** [reply]
Thanks for replying back.
So you mean there is actually no dedicated pin like PPM on the board? How can I help out as a volunteer?

**Reply by ajjjjjjjj** [reply]
You may implement it on any of the serial enabled pins. See STM32F072 datasheet and PCB, but those near unpopulated module on the right side of the PCB are a good candidate.

**Reply by FRPProject** [reply]
So then you saying that I have to just define an unused pin on STM32F072 in the target lists here: 

https://github.com/OpenI6X/opentx/blob/master/radio/src/targets/flysky/hal.h

and then perhaps, send the SBUS signal to that predefined pin. Is that correct?

**Reply by ajjjjjjjj** [reply]
Define and configure TX and RX pins as serial and then see in OpenTX how other board targets configure it and try to do the same on STM32F0 keeping in mind that we have pin inversion in HW.

**Reply by FRPProject** [reply]
Sounds like we have a plan!

**Reply by FRPProject** [reply]
I found a few good leads of STM32 parsing SBUS signal routine. Seems promising, hopefully others can join to help out as well.

**OpenTX SBUS configuration on Tranis:**
- https://github.com/opentx/opentx/blob/2.3/radio/src/targets/taranis/hal.h

**Other sources:**
- https://www.codetd.com/en/article/13078381
- https://github.com/bolderflight/sbus
- https://quadmeup.com/generate-s-bus-with-arduino-in-a-simple-way/
- https://openrcforums.com/forum/viewtopic.php?t=6109
- https://github.com/MikeBland/SbusToPWM
- https://github.com/opentx/opentx/issues/8119

**Reply by ajjjjjjjj** [reply]
Nice, i found this one, with code examples: https://blog.katastros.com/a?ID=01000-4a28701c-2e84-4f1c-ad0b-9b749c0d1ca0

**Reply by FRPProject** [reply]
Great! Still I am not sure which pin would be the best to configure the SBUS on the FSI6X PCB. Any suggestion?

**Reply by ajjjjjjjj** [reply]
Check those RX and TX pins on unpopulated module on the right side of PCB. Don't know which of the gpios are connected. You may want to try to figure it out with multimeter and then check in datasheet what is possible on those pins.

**Reply by dgehriger** [reply]
@ajjjjjjjj  / @FRPProject : did you make progress on implementing serial communication with an external RF module? If not, what are your current conclusions (which serial pin to use, ...)?

**Reply by FRPProject** [reply]
No I didn't make much progress, I decided to get ELRS module.

**Reply by ajjjjjjjj** [reply]
@dgehriger you can use any serial available on the board, maybe reuse TX2 pad - same as CRSF.

**Reply by dgehriger** [reply]
@ajjjjjjjj  Thanks! Just to clear up some confusion on my side: the multi protocol module doesn't expose a full serial (rx/tx) but a "smart serial" using half duplex? Correct ?
